app/src/main/java/com/example/taskyy/ui/viewmodels/RegisterViewModel.kt

```diff
@@ -21,21 +24,24 @@ class RegisterViewModel @Inject constructor(
             is RegisterEvent.OnNameChanged -> state = state.copy(name = event.name)
             is RegisterEvent.OnEmailChanged -> state = state.copy(email = event.email)
             is RegisterEvent.OnPasswordChanged -> state = state.copy(password = event.password)
-            is RegisterEvent.OnGetStartedClick -> register()
+            is RegisterEvent.OnGetStartedClick -> register(context = event.applicationContext)
             is RegisterEvent.OnTogglePasswordVisibility -> state = state.copy(isPasswordVisible = event.isPasswordVisible)
-                   }
+        }
     }
 
-    private fun register() {
-        authRepository.registerUser(name = state.name, password = state.password, email = state.email)
+    private fun register(context: Context) {
+        val user = User(name = state.name, password = state.password, email = state.email)
+        val dao = TaskyyDatabase.getDatabase(context = context).userDao()
```

Comment:
Here I'd recommend to get into proper dependency injection, so you would pass a reference to this DB (or better just the DAO you need here) in the constructor of the ViewModel, so it remains flexible what you pass for that.
A good starting point to get into that is my course on Dagger-Hilt:
https://www.youtube.com/watch?v=bbMsuI2p1DQ